app/models/user.rb

```diff
@@ -94,10 +96,22 @@ def self.find_for_google_oauth2(access_token)
     user
   end
 
+  # to refactor, could be single oauth method to begin with
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
```

Comment:
I think uid is the correct option here than email

Reply:
I'm fine with using UID here!